src/Object/Photo.php

```diff
@@ -0,0 +1,1206 @@
+				if (file_exists($filename)) {
+					unlink($filename);
+				}
+				$filename = $basepath.'/photo/'.$hash.'-5.'.$img->getExt();
+				if (file_exists($filename)) {
+					unlink($filename);
+				}
+				$filename = $basepath.'/photo/'.$hash.'-6.'.$img->getExt();
+				if (file_exists($filename)) {
+					unlink($filename);
+				}
+			}
+		} else {
+			$photo_failure = true;
+		}
+	
+		if ($photo_failure && $quit_on_error) {
+			return false;
+		}
+	
+		if ($photo_failure) {
+			$photo = System::baseUrl() . '/images/person-175.jpg';
+			$thumb = System::baseUrl() . '/images/person-80.jpg';
+			$micro = System::baseUrl() . '/images/person-48.jpg';
+		}
+	
+		return(array($photo, $thumb, $micro));
+	}
+
+	/**
+	 * @param string $url url
+	 * @return object
+	 */
+	public function getPhotoInfo($url)
+	{
+		$data = array();
+	
+		$data = Cache::get($url);
+	
+		if (is_null($data) || !$data || !is_array($data)) {
+			$img_str = fetch_url($url, true, $redirects, 4);
+			$filesize = strlen($img_str);
+	
+			if (function_exists("getimagesizefromstring")) {
+				$data = getimagesizefromstring($img_str);
+			} else {
+				$tempfile = tempnam(get_temppath(), "cache");
+	
+				$a = get_app();
+				$stamp1 = microtime(true);
+				file_put_contents($tempfile, $img_str);
+				$a->save_timestamp($stamp1, "file");
+	
+				$data = getimagesize($tempfile);
+				unlink($tempfile);
+			}
+	
+			if ($data) {
+				$data["size"] = $filesize;
+			}
+	
+			Cache::set($url, $data);
+		}
+	
+		return $data;
+	}
+
+	/**
+	 * @param integer $width  width
+	 * @param integer $height height
+	 * @param integer $max    max
+	 * @return array
+	 */
+	public function scaleImageTo($width, $height, $max)
+	{
+		$dest_width = $dest_height = 0;
+	
+		if ((!$width) || (!$height)) {
+			return false;
+		}
+	
+		if ($width > $max && $height > $max) {
+			// very tall image (greater than 16:9)
+			// constrain the width - let the height float.
+	
+			if ((($height * 9) / 16) > $width) {
+				$dest_width = $max;
+				$dest_height = intval(($height * $max) / $width);
+			} elseif ($width > $height) {
+				// else constrain both dimensions
+				$dest_width = $max;
+				$dest_height = intval(($height * $max) / $width);
+			} else {
+				$dest_width = intval(($width * $max) / $height);
+				$dest_height = $max;
+			}
+		} else {
+			if ($width > $max) {
+				$dest_width = $max;
+				$dest_height = intval(($height * $max) / $width);
+			} else {
+				if ($height > $max) {
+					// very tall image (greater than 16:9)
+					// but width is OK - don't do anything
+	
+					if ((($height * 9) / 16) > $width) {
+						$dest_width = $width;
+						$dest_height = $height;
+					} else {
+						$dest_width = intval(($width * $max) / $height);
+						$dest_height = $max;
+					}
+				} else {
+					$dest_width = $width;
+					$dest_height = $height;
+				}
+			}
+		}
+		return array("width" => $dest_width, "height" => $dest_height);
+	}
+
+	/**
+	 * @brief This function doesn't seem to be used
```

Comment:
If the function isn't used anywhere, please scrap it. We will be able to find it again in the git history if needed. There already is a `Photo::store()` function anyway.

Reply:
It is used in the addon "fromgplus".

Reply:
What is the difference with `Photo::store()` ?

Reply:
Good question. I guess we have to check the differences by comparing the functionality.